karibu-dsl/src/main/kotlin/com/github/mvysny/karibudsl/v10/KLitRendererTagsDsl.kt

```diff
@@ -0,0 +1,206 @@
+
+            else -> children.joinToString(separator = "")
+        }
+
+    }
+
+    override fun Property<TSource>.unaryPlus() {
+        children.add(
+            KLitRendererTagsBuilder(
+                litRendererBuilder,
+                propertyName = (litRendererBuilder as KLitRendererBuilder).propertyName(this)
+            )
+        )
+    }
+
+    override fun KLitRendererBuilderA.Function<TSource>.unaryPlus() {
+        children.add(
+            KLitRendererTagsBuilder(
+                litRendererBuilder,
+                propertyName = (litRendererBuilder as KLitRendererBuilder).functionName(this)
+            )
+        )
+    }
+
+    override fun String.unaryPlus() {
+        children.add(
+            KLitRendererTagsBuilder(
+                litRendererBuilder,
+                propertyName = this
+            )
+        )
+    }
+
+    private fun addTag(
+        tagName: String,
+        attributes: Sequence<KLitRendererAttribute>,
+        block: KLitRendererTagsBuilderA<TSource>.() -> Unit,
+        selfClosing: Boolean = false,
+    ) {
+        children.add(
+            KLitRendererTagsBuilder(
+                litRendererBuilder,
+                tagName,
+                attributes.joinToString(separator = " "),
+                selfClosing = selfClosing
+            ).apply(block)
+        )
+    }
+
+    override fun horizontalLayout(
+        vararg attributes: KLitRendererAttribute,
+        block: KLitRendererTagsBuilderA<TSource>.() -> Unit
+    ) {
+        addTag("vaadin-horizontal-layout", attributes.asSequence(), block)
+    }
+
+    override fun verticalLayout(
+        vararg attributes: KLitRendererAttribute,
+        block: KLitRendererTagsBuilderA<TSource>.() -> Unit
+    ) {
+        addTag("vaadin-vertical-layout", attributes.asSequence(), block)
+    }
+
+    override fun span(
+        vararg attributes: KLitRendererAttribute,
+        block: KLitRendererTagsBuilderA<TSource>.() -> Unit
+    ) {
+        addTag("span", attributes.asSequence(), block)
+    }
+
+    override fun div(
+        vararg attributes: KLitRendererAttribute,
+        block: KLitRendererTagsBuilderA<TSource>.() -> Unit
+    ) {
+        addTag("div", attributes.asSequence(), block)
+    }
+
+    override fun anchor(
+        vararg attributes: KLitRendererAttribute,
+        block: KLitRendererTagsBuilderA<TSource>.() -> Unit
+    ) {
+        addTag("a", attributes.asSequence(), block)
+    }
+
+    override fun icon(
+        vararg attributes: KLitRendererAttribute,
+        block: KLitRendererTagsBuilderA<TSource>.() -> Unit
+    ) {
+        addTag("vaadin-icon", attributes.asSequence(), block)
+    }
+
+    override fun button(
+        vararg attributes: KLitRendererAttribute,
+        block: KLitRendererTagsBuilderA<TSource>.() -> Unit
+    ) {
+        addTag("vaadin-button", attributes.asSequence(), block)
+    }
+
+    /**
+     * "<img src=${item.image} />"
```

Comment:
It's hard to read this kind of kdoc. Better way: Creates an `<img src="..." />` tag. Intended usage: xyz, will output: abc. Note: this tag is self-closing: [selfClosing] is set to true.

Also, please avoid using custom formatting such as `---`: please only use the supported kdoc syntax as present here: https://kotlinlang.org/docs/kotlin-doc.html